lib/socket.io/router.js

```diff
@@ -52,18 +52,25 @@ Router.prototype.middleware = function () {
     socket._onevent = socket.onevent;
     socket.onevent = function (packet) {
       var args = packet.data || [];
+
       if (!args.length) {
         debug('event args not exist');
         return socket._onevent(packet);
       }
 
+      if(null != packet.id) {
+        debug('attaching ack callback to event')
+        args.push(socket.ack(packet.id))
+      }
+      
       context.event = args[0];
       context.data = args.slice(1);
 
-      co.wrap(gen).call(context, function (err) {
-        if (err) console.error(err.stack);
-        //TODO: error
```

Comment:
It looks like here you need to handle the error, or at least deal with it. Right now you just log it and continue.